### PR TITLE
fix: check environment when doing pre-upgrade with Helm

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#7571

#### What this PR does / why we need it:

Check if the environment settings are achieved when doing pre-upgrade job with Helm, ref: [environment_check.sh](https://github.com/longhorn/longhorn/blob/master/scripts/environment_check.sh)

### CODE REVIEW

File: chart/templates/preupgrade-job.yaml [hunk 1]
```diff
@@ -20,14 +20,27 @@ spec:
...
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+      volumes:
+      - name: proc
+        hostPath:
```
Comment (derekbit) [inline chart/templates/preupgrade-job.yaml:42]: Adding  a description for the `/proc` and `privileged: true` in the commit message  is appreciated.

Comment (mantissahz) [reply chart/templates/preupgrade-job.yaml:42]: Added in the commit message. Thanks

**Comment by mantissahz** [comment]
Moved the check to the node controller.

**Comment by mantissahz** [comment]
For checking iscsiadm/open-iscsi installed, we need this PR to be reopened.

related: https://github.com/longhorn/longhorn/issues/8974

**Review by derekbit — APPROVED** [review]
LGTM

**Pull request merged**